sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/ecert-part-time-process-integration.scheduler.ts

```diff
@@ -3,48 +3,67 @@ import { ECertFileHandler } from "@sims/integrations/esdc-integration";
 import { QueueService } from "@sims/services/queue";
 import { QueueNames } from "@sims/utilities";
 import { Job, Queue } from "bull";
-import { QueueProcessSummary } from "../../../models/processors.models";
 import { BaseScheduler } from "../../base-scheduler";
-import { ESDCFileResult } from "../models/esdc.models";
+import { PartTimeCalculationProcess } from "@sims/integrations/services/disbursement-schedule/e-cert-calculation";
+import {
+  InjectLogger,
+  LoggerService,
+  ProcessSummary,
+} from "@sims/utilities/logger";
+import {
+  getSuccessMessageWithAttentionCheck,
+  logProcessSummaryToJobLogger,
+} from "../../../../utilities";
 
 @Processor(QueueNames.PartTimeECertIntegration)
 export class PartTimeECertProcessIntegrationScheduler extends BaseScheduler<void> {
   constructor(
     @InjectQueue(QueueNames.PartTimeECertIntegration)
     schedulerQueue: Queue<void>,
     queueService: QueueService,
+    private readonly partTimeCalculationProcess: PartTimeCalculationProcess,
     private readonly eCertFileHandler: ECertFileHandler,
   ) {
     super(schedulerQueue, queueService);
   }
 
   /**
-   * Process Part-Time disbursements available to be sent to ESDC.
+   * Process part-Time disbursements available to be sent to ESDC.
    * Consider any record that is scheduled in upcoming days or in the past.
    * @params job job details.
    * @returns result of the file upload with the file generated and the
    * amount of records added to the file.
    */
   @Process()
-  async processPartTimeECert(job: Job<void>): Promise<ESDCFileResult> {
-    const summary = new QueueProcessSummary({
-      appLogger: this.logger,
-      jobLogger: job,
-    });
-    await summary.info(
-      `Processing E-Cert Part-time integration job ${job.id} of type ${job.name}.`,
-    );
-    await summary.info("Sending Part-time E-Cert File...");
-    const uploadPartTimeResult =
-      await this.eCertFileHandler.generatePartTimeECert();
-    await summary.info("E-Cert Part-time file sent.");
-    await this.cleanSchedulerQueueHistory();
-    await summary.info(
-      `Completed E-Cert Part-time integration job ${job.id} of type ${job.name}.`,
-    );
-    return {
-      generatedFile: uploadPartTimeResult.generatedFile,
-      uploadedRecords: uploadPartTimeResult.uploadedRecords,
-    };
+  async processPartTimeECert(job: Job<void>): Promise<string[]> {
```

Comment:
I was wondering if we need to define `void` as the generic type here. TypeScript doesn't complain about it.

Reply:
It does not complain but it will consider the type as `any` and the idea is to avoid it 😉